src/library/FIFO.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A First-In-First-Out (FIFO) class that maintains a queue of
+ * elements.  New elements are added to the tail, and the oldest
+ * element is removed from the head when the FIFO is full.  If the
+ * size of the queue is 1, it will always contain the newest element
+ * that was inserted.
+ * @class
+ */
+
+class FIFO {
+  
+  /**
+   * @constructor Constructs a FIFO (First-In-First-Out) object.
+   * @param {Number} size The size of the FIFO.
+   */
+  constructor(size) {
+    /** @private @const {!size} The size of the FIFO */
+    this.size_ = size;
+    /** @private @const {!data} The buffer data of the FIFO */
+    this.data_ = new Float32Array(size);
+    /** @private {!head} The index of the first element in the FIFO
+     * */
+    this.head_ = 0;
+    /** @private {!tail} The index of the last element in the FIFO
+     * */
+    this.tail_ = 0;
+    /** @private {!count} The current size of the FIFO */
+    this.count_ = 0;
+  }
+
+  /**
+   * Add a new value to the tail index of the queue. If the FIFO is
+   * full, it removes the element at the head index (the oldest).
+   * @param {Number} value The new value to be added to the FIFO.
+   */
+  add(value) {
```

Comment:
It looks like there's no way to get the data back from the FIFO. Did I get that right?

Also for the consistency, I would recommend using `push()` instead of `add()`.